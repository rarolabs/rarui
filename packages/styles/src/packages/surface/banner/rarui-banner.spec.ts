import { banner } from ".";

describe("Banner Styles", () => {
  it("should generate correct default styles", () => {
    const className = banner.classnames.banner();
    expect(className).toMatchSnapshot();
  });

  it("should generate correct styles for each appearance variant", () => {
    const appearances = [
      "brand",
      "info",
      "neutral",
      "inverted",
      "warning",
      "error",
    ] as const;

    appearances.forEach((appearance) => {
      const className = banner.classnames.banner({ appearance });
      expect(className).toMatchSnapshot(`${appearance} appearance`);
    });
  });

  it("should generate correct styles for floating variant", () => {
    const className = banner.classnames.banner({ floating: true });
    expect(className).toMatchSnapshot("floating true");
  });

  it("should generate correct compound variants", () => {
    const compoundVariants = [
      { appearance: "neutral", floating: true },
      { appearance: "inverted", floating: true },
    ];

    compoundVariants.forEach((variants) => {
      const className = banner.classnames.banner(variants as any);
      expect(className).toMatchSnapshot(
        `${variants.appearance} ${variants.floating} compound`,
      );
    });
  });
});
