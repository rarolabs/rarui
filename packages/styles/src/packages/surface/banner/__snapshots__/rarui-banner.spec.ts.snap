// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner Styles should generate correct compound variants: inverted true compound 1`] = `"rarui-banner__1wgs6s20 rarui-banner_appearance_inverted__1wgs6s24 rarui-banner_floating_true__1wgs6s27 rarui-banner__compound_1__1wgs6s29"`;

exports[`Banner Styles should generate correct compound variants: neutral true compound 1`] = `"rarui-banner__1wgs6s20 rarui-banner_appearance_neutral__1wgs6s23 rarui-banner_floating_true__1wgs6s27 rarui-banner__compound_0__1wgs6s28"`;

exports[`Banner Styles should generate correct default styles 1`] = `"rarui-banner__1wgs6s20"`;

exports[`Banner Styles should generate correct styles for each appearance variant: brand appearance 1`] = `"rarui-banner__1wgs6s20 rarui-banner_appearance_brand__1wgs6s21"`;

exports[`Banner Styles should generate correct styles for each appearance variant: error appearance 1`] = `"rarui-banner__1wgs6s20 rarui-banner_appearance_error__1wgs6s26"`;

exports[`Banner Styles should generate correct styles for each appearance variant: info appearance 1`] = `"rarui-banner__1wgs6s20 rarui-banner_appearance_info__1wgs6s22"`;

exports[`Banner Styles should generate correct styles for each appearance variant: inverted appearance 1`] = `"rarui-banner__1wgs6s20 rarui-banner_appearance_inverted__1wgs6s24"`;

exports[`Banner Styles should generate correct styles for each appearance variant: neutral appearance 1`] = `"rarui-banner__1wgs6s20 rarui-banner_appearance_neutral__1wgs6s23"`;

exports[`Banner Styles should generate correct styles for each appearance variant: warning appearance 1`] = `"rarui-banner__1wgs6s20 rarui-banner_appearance_warning__1wgs6s25"`;

exports[`Banner Styles should generate correct styles for floating variant: floating true 1`] = `"rarui-banner__1wgs6s20 rarui-banner_floating_true__1wgs6s27"`;
